otherlibs/systhreads/st_stubs.c

```diff
@@ -711,12 +711,12 @@ CAMLprim value caml_thread_yield(value unit)        /* ML */
 {
   if (atomic_load_acq(&Thread_main_lock.waiters) == 0) return Val_unit;
 
-  caml_process_pending_signals();
+  caml_process_pending_actions();
```

Comment:
I'm not convinced this matches the behaviour in 4.x. If we look at https://github.com/ocaml/ocaml/blob/4.14/otherlibs/systhreads/st_stubs.c#L734-L739 we explicitly only deal with signals in `caml_thread_yield`.

Reply:
I agree, I think my reasoning was that in theory with new polling locations, any signal handler could execute other actions and so the distinction is no longer relevant. But the reasoning was probably flawed and in any case not for this PR which is meant to be side-effect-free.

Some documentation comment erroneously disappeared in multicore, which encouraged to keep consistency with caml_enter/leave_blocking_section, which was useful information in this case. I restored the comment.

In light of this comment, the line `Current_thread = st_tls_get(Thread_key);` was added in caml_thread_enter_blocking_section but not in caml_thread_yield, so there is a now a discrepancy, please have a look.

Reply:
Yes, this is a good point. @Engil do you happen to know how we ended up with this? trunk seems to update the current thread when entering a blocking section: https://github.com/ocaml/ocaml/blob/1f9c958e8d077c4e2f67a122beccccc831e5c394/otherlibs/systhreads/st_stubs.c#L210 but this doesn't seem to make sense? I think you must be the current thread in order to call `caml_thread_enter_blocking_section`.

(4.14 doesn't do this: https://github.com/ocaml/ocaml/blob/4.14/otherlibs/systhreads/st_stubs.c#L230 )

Reply:
I think this is an oversight and is not required.
Weird interactions with backup threads or else do not seem likely at all (since a thread going through `caml_thread_enter_blocking_section` is the thread actually holding the lock and the thread that has been executing code.).
I think it is safe to remove this.

Reply:
Ok, thanks. I'll make a PR doing so.